### ISSUE TITLE
Inconsistent filter options between cli commands

### ISSUE THREAD

**Issue by thomassuckow** [docker/cli#833]
**Description**

Commands like
`docker image ls`
and
`docker image prune`

have wildly different filter options

**Describe the results you received:**

ls has `label` `before` `since` `reference` `dangling`
prune has `label` (Which apparently doesn't even work on docker for mac) and `until` and a "-a" flag which is like `dangling`

**Describe the results you expected:**

I expect there to be a consistent set of filter options on all commands.

**Output of `docker version`:**

$ docker version
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:03:51 2017
 OS/Arch:	darwin/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:12:29 2017
  OS/Arch:	linux/amd64
  Experimental:	true

**Output of `docker info`:**

$ docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 30
Server Version: 17.12.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 94
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.60-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: YFR2:2A3B:JXTC:KQ5F:H7DB:NAD3:VOJK:U4H2:IDYF:EXOW:A2VE:D3JP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 22
 Goroutines: 41
 System Time: 2018-01-25T18:24:14.397347924Z
 EventsListeners: 2
HTTP Proxy: docker.for.mac.http.internal:3128
HTTPS Proxy: docker.for.mac.http.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by DenKoren** [comment]
This is true not only for images, but is seen across 'prune' commands for various instances as well.

For example, 
`docker image prune`
supports 'until' flag, while 
`docker volume prune`
doesn't.

**Comment by sevillaarvin** [comment]
Or how about `docker images` which lists all the images. But `docker volume` does not list all volumes. It needs `docker volume ls`.

**Comment by thomassuckow** [comment]
docker images is legacy. The `docker image ls` command is the modern comparison.
